### ISSUE TITLE
Supplying the same argument twice doesn't cause error

### ISSUE THREAD

**Issue by yurivict** [Taywee/args#117]
These arguments:
```
args::ValueFlag<std::string>  a1  (parser, "a1", "a1", {'d', "arg1"});
args::ValueFlag<std::string>  a2  (parser, "a2", "a2", {'d', "arg2"});
```
aren't valid since both are activated with ```-d```, but ```args``` doesn't fail.

**Comment by Taywee** [comment]
Yes.  That is a shortcoming that is difficult to work around, given the generality of this library (argument groups may be arbitrarily nested and combinations of them are similarly arbitrary).  I'd be glad to have the library catch this case and raise an error, but I'm not planning on writing this change myself.  I'd happily accept a pull request with an appropriate test case that fixes it, though.
